BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -0,0 +1,1041 @@
+            float rotDistance = Vector3.Angle(vessel.ReferenceTransform.up, toEscape) * Mathf.Deg2Rad;
+            float timeToRotate = SolveTime(rotDistance / 2, maxAngularAcceleration.magnitude) * 2;
+            float timeToDisplace = SolveTime(minRange - toTarget.magnitude, maxAcceleration, Vector3.Dot(relVel * -1, toEscape));
+            float timeToEscape = timeToRotate * 2 + timeToDisplace;
+
+            Vector3 drift = AIUtils.PredictPosition(toTarget, relVel, Vector3.zero, timeToEscape);
+            bool manualEscape = drift.magnitude < minRange;
+
+            return manualEscape;
+        }
+
+        private Vector3 ToClosestApproach(Vector3 toTarget, Vector3 relVel, float minRange)
+        {
+            Vector3 relVelInverse = targetVessel.GetObtVelocity() - vessel.GetObtVelocity();
+            float timeToIntercept = AIUtils.TimeToCPA(toTarget, relVelInverse, Vector3.zero, 9999);
+
+            // Minimising the target closest approach to the current closest approach prevents
+            // ships that are targeting each other from fighting over the closest approach based on their min ranges.
+            // todo: allow for trajectory fighting if fuel is high.
+            Vector3 actualClosestApproach = toTarget + Displacement(relVelInverse, Vector3.zero, timeToIntercept);
+            float actualClosestApproachDistance = actualClosestApproach.magnitude;
+
+            // Get a position that is laterally offset from the target by our desired closest approach distance.
+            Vector3 rotatedVector = Vector3.ProjectOnPlane(relVel, toTarget.normalized).normalized;
+
+            // Lead if the target is accelerating away from us.
+            if (Vector3.Dot(targetVessel.acceleration.normalized, toTarget.normalized) > 0)
+                toTarget += Displacement(Vector3.zero, toTarget.normalized * Vector3.Dot(targetVessel.acceleration, toTarget.normalized), Mathf.Min(timeToIntercept, 999));
+
+            Vector3 toClosestApproach = toTarget + (rotatedVector * Mathf.Clamp(actualClosestApproachDistance, minRange, weaponManager.gunRange * 0.5f));
+
+            // Need a maximum angle so that we don't end up going further away at close range.
+            toClosestApproach = Vector3.RotateTowards(toTarget, toClosestApproach, 22.5f * Mathf.Deg2Rad, float.MaxValue);
+
+            return toClosestApproach;
+        }
+
+        internal void RestoreReferenceTransform()
+        {
+            vessel.SetReferenceTransform(originalReferenceTransform);
+        }
+
+        #endregion
+
+        #region Utils
+        public static Vector3 FromTo(Vessel v1, Vessel v2)
+        {
+            return v2.transform.position - v1.transform.position;
+        }
+
+        public static Vector3 RelVel(Vessel v1, Vessel v2)
+        {
+            return v1.GetObtVelocity() - v2.GetObtVelocity();
+        }
+
+        public static Vector3 RelAccel(Vessel v1, Vessel v2)
+        {
+            return v1.acceleration - v2.acceleration;
```

Comment:
These should probably use `.acceration_immediate`, but it probably doesn't make much difference here... especially since `RelAccel` doesn't appear to be used anywhere.

There's several unused functions here too.